### ISSUE TITLE
[Bug] Track repeats three times in Playback History

### ISSUE THREAD

**Issue by snppls** [jmshrv/finamp#688]
On iOS 17.3.1 and Finamp 0.9.5 every song played appears exactly three times in the Playback History rather than once.

![IMG_0367](https://github.com/jmshrv/finamp/assets/156883144/8559c629-8476-488b-af2f-1dedf5bfd4aa)

**Comment by Chaphasilor** [comment]
I guess that answers the question if anyone else uses the playback history :P

Can you get me some logs? On Android it works fine, so I think this is a platform-specific issue.  

![Screenshot_20240409-112358.png](https://github.com/jmshrv/finamp/assets/18015147/f2c8fc3b-6471-4294-8584-e02a6f32c207)

Never-ending story with just_audio and playback tracking 😅

**Comment by snppls** [comment]
Here is a snippet of my recent logs. Attached is the full log - its pretty massive. 
[logs.txt](https://github.com/jmshrv/finamp/files/14917831/logs.txt)

[QueueService/FINEST] 2024-04-09 21:13:17.164431: Saved new periodic queue previous:[56d7ca4a7389729ef28d58727416cf0f, b606ae871125cf62cbbd082398332e06, 0c57f7d3e07d15fa2cb65e0a7ef55334, a4f5c0de71ed87bfa4d090ef3a41318b, 12ba7edd5b807cec2322296714080bdd] current:526275b2974c05ecf0228eca6d631a57 seek:10440 next:[] queue:[85a3f66bbf4d37a03008991f6010400e, 4f962317991ba3d4c133aaf2abceb276, ce3ea4c785737e2070e49fd4726fe155, 6be3ea8c12331e2920f18b11aec03f29, c1b9edde86c31994d11ce11580e63830, ef2b33d824b1378f272ea5167203602c, d7f5575dba5191da7bfc975a8f458999, ce0e9fc4c7070c9374d58ad36430cd11, 645679c6462bb3391efb3e826b86a21d, cbc6c0133d8f60476c6d4e7aacc2bf39, 4cfd90b1ec48dec8c54f16dc7311ef44, 08e290f043f3fb675c4a850bf5a8e493, d6779ec083098309210025541e13211a, 8985a6f25ce4282b287cb06a1443b898, 2476e901c70435e84891d3b142cc9aed]
[Chopper/INFO] 2024-04-09 21:13:17.216018: --> POST BASEURL/Sessions/Playing/Progress
		Content-Type: application/json; charset=utf-8
		Authorization: MediaBrowser UserId="47a30ddf28ee439ca0fd8dddfbc4c6dc", Token="TOKEN", Client="Finamp", Device="iPhone", DeviceId="B8CD80B9-8385-4799-8D79-ED0376F2D314", Version="0.9.5"
		content-length: 520
		
		{"CanSeek":true,"Item":null,"ItemId":"526275b2974c05ecf0228eca6d631a57","SessionId":null,"MediaSourceId":null,"AudioStreamIndex":null,"SubtitleStreamIndex":null,"IsPaused":true,"IsMuted":false,"PositionTicks":104400000,"PlaybackStartTimeTicks":17126364419930000,"VolumeLevel":56,"Brightness":null,"AspectRatio":null,"PlayMethod":"Transcode","LiveStreamId":null,"PlaySessionId":"56d1d47b-e311-4fec-a6d4-c2c577ad43f0","RepeatMode":"RepeatAll","NowPlayingQueue":null,"PlaylistItemId":"6f5c0c5e-8e6c-46a1-ab5c-1e3653778fa1"}
		--> END POST
[downloadsService/INFO] 2024-04-09 21:13:17.221662: App returning from background, restarting downloads.
[downloadsService/INFO] 2024-04-09 21:13:17.222006: Attempting to restart queues.
[SyncBuffer/INFO] 2024-04-09 21:13:17.222947: All syncs complete.
[DeleteBuffer/INFO] 2024-04-09 21:13:17.223032: All deletes complete.
[IsarTaskQueue/INFO] 2024-04-09 21:13:17.223327: All downloads enqueued.
[ThemeProvider/FINEST] 2024-04-09 21:13:17.260322: Accent color: Color(0xff113652)
[AtContrast/INFO] 2024-04-09 21:13:17.261018: Calculated contrast in 1 steps
[Chopper/INFO] 2024-04-09 21:13:17.317405: <-- 204 No Content POST BASEURL/Sessions/Playing/Progress
		x-response-time-ms: 59
		date: Tue, 09 Apr 2024 11:13:16 GMT
		server: Kestrel
		content-length: 0
		<-- END HTTP
[JellyfinApiHelper/FINE] 2024-04-09 21:13:19.276227: Getting children of Solange
[JellyfinApiHelper/FINE] 2024-04-09 21:13:19.277157: Getting children of Solange
[JellyfinApiHelper/FINE] 2024-04-09 21:13:19.277396: Getting children of Solange
[JellyfinApiHelper/FINE] 2024-04-09 21:13:21.742214: Getting children of Music
[JellyfinApiHelper/FINE] 2024-04-09 21:13:21.742740: Getting children of Music
[JellyfinApiHelper/FINE] 2024-04-09 21:13:21.743029: Getting items.
[QueueService/FINEST] 2024-04-09 21:13:22.981522: Saved new periodic queue previous:[56d7ca4a7389729ef28d58727416cf0f, b606ae871125cf62cbbd082398332e06, 0c57f7d3e07d15fa2cb65e0a7ef55334, a4f5c0de71ed87bfa4d090ef3a41318b, 12ba7edd5b807cec2322296714080bdd] current:526275b2974c05ecf0228eca6d631a57 seek:10440 next:[] queue:[85a3f66bbf4d37a03008991f6010400e, 4f962317991ba3d4c133aaf2abceb276, ce3ea4c785737e2070e49fd4726fe155, 6be3ea8c12331e2920f18b11aec03f29, c1b9edde86c31994d11ce11580e63830, ef2b33d824b1378f272ea5167203602c, d7f5575dba5191da7bfc975a8f458999, ce0e9fc4c7070c9374d58ad36430cd11, 645679c6462bb3391efb3e826b86a21d, cbc6c0133d8f60476c6d4e7aacc2bf39, 4cfd90b1ec48dec8c54f16dc7311ef44, 08e290f043f3fb675c4a850bf5a8e493, d6779ec083098309210025541e13211a, 8985a6f25ce4282b287cb06a1443b898, 2476e901c70435e84891d3b142cc9aed]

**Comment by Chaphasilor** [comment]
Okay, firstly, thanks a ton for the sponsorship! I really appreciate it <3

Regarding your issue:  
The logs you sent me start at 2pm and don't seem to contain the tracks shown in the screenshot. Could you try playing a few songs, taking the screenshot, and then extracting the logs (making sure the app doesn't close before, logs aren't persisted yet)?

I also can't see anything suspicious in the logs, but that is probably due to a lack of logging on my side. I'll try to add more verbose logging to the next update to better track this down.

Lastly, does this really happen with all tracks? Listening all day today I only saw this a single time, and I'm not even sure that really was an error (might've just been me pausing for a while or something).  
Could you please take a look at your Jellyfin dashboard to see if the tracks are also reported to the server multiple times, or if it's just a local issue? It should look something like this (with each track being reported as "is playing" first, and then as "has finished playing"). There shouldn't be multiple reports of a track, unless you actually rewound/looped it).

**Comment by snppls** [comment]
No problems, I appreciate your hard work.

Attached are my most recent logs and a screen recording to illustrate the issue. The problem seems to be with every track. On rare occasions the track with appear twice rather than three times in playback history. 

[logs.txt](https://github.com/jmshrv/finamp/files/14925787/logs.txt)

https://github.com/jmshrv/finamp/assets/156883144/0574ecda-249c-408c-8259-cd0e36db0d23

**Comment by Chaphasilor** [comment]
I'm guessing this is still a problem?

**Comment by snppls** [comment]
Yes, still a problem. The same issue persists as of 0.9.7 on iOS.

**Comment by 9p4** [comment]
Can confirm it's a problem, it causes ListenBrainz to double-report playbacks—once on start and once on track end.
